Additional review and application of QE feedback to the  Datasource guide (#35843)
quarkusio/quarkus: docs/src/main/asciidoc/datasource.adoc

Applications use datasources to access relational databases.
Quarkus provides a unified configuration model to define datasources for Java Database Connectivity (JDBC) and Reactive database drivers.

Quarkus uses link:https://agroal.github.io/[Agroal] and link:https://vertx.io/[Vert.x] to provide high-performance, scalable data source connection pooling for JDBC and reactive drivers.
The `jdbc-\*` and `reactive-*` extensions provide build time optimizations and integrate configured data sources with Quarkus features like security, health checks, and metrics.

> For more information on consuming and using a reactive datasource, consult the Quarkus xref:reactive-sql-clients.adoc[Reactive SQL clients] guide.

Reviewer: ```suggestion
For more information about consuming and using a reactive datasource, see the Quarkus xref:reactive-sql-clients.adoc[Reactive SQL clients] guide.
```

<1> This configuration value is only required if there is more than one database extension on the classpath.

If only one viable extension is available, Quarkus assumes this is the correct one.
When you add a driver to the test scope, Quarkus automatically includes the specified driver in testing.

==== JDBC connection pool size adjustment

> To protect your database from overloading during the transaction load peaks, size the pool adequately to throttle the database load.

Reviewer 2: ```suggestion
To protect your database from overloading during load peaks, size the pool adequately to throttle the database load.
```

<1> This configuration value is only required if there is more than one database extension on the classpath.

If only one viable extension is available, Quarkus assumes this is the correct one.
When a driver has been added to the test scope, Quarkus automatically includes the driver in testing.

==== JDBC connection pool size adjustment

To protect your database from overloading during the transaction load peaks, size the pool adequately to throttle the database load.
The proper size always depends on many factors, such as the number of parallel application users or the nature of the workload.

> Be aware that this pool size adjustment will cause some requests to time out while waiting for a connection.

Reviewer 3: ```suggestion
Be aware that setting the pool size too low may cause some requests to time out while waiting for a connection.
```

<1> This configuration value is only required if there is more than one database extension on the classpath.

If only one viable extension is available, Quarkus assumes this is the correct one.

> When a driver has been added to the test scope, Quarkus automatically includes the driver in testing.

Reviewer: Make active. If the developer always does this action. Otherwise, leave as-is and close this comment.

```suggestion
When you add a driver to the test scope, Quarkus automatically includes the specified driver in testing.
```

==== JDBC connection pool size adjustment

To protect your database from overloading during the transaction load peaks, size the pool adequately to throttle the database load.

> The proper size always depends on many factors, such as the number of parallel application users or the nature of the workload.

Reviewer: Specificity:
```suggestion
The optimal pool size depends on many factors, such as the number of parallel application users or the nature of the workload.
```

Be aware that setting the pool size too low might cause some requests to time out while waiting for a connection.

> For more information about pool size adjustment properties, see the xref:jdbc-configuration[] section.

Reviewer: For ease of doc maintenance and automatic title updates in links, use `<<section-ID>>` instead of xrefs.

```suggestion
For more information about pool size adjustment properties, see the <<jdbc-configuration>> section.
```

_The downstreamdoc.java postprocessor build job converts these to xrefs for PV2 consumption._

In the above example, the type is `AgroalDataSource`, a `javax.sql.DataSource` subtype.
Because of this, you can also use `javax.sql.DataSource` as the injected type.

==== Reactive datasource

> Quarkus offers several reactive clients for a use with reactive datasource.

Reviewer: Fix typo:

```suggestion
Quarkus offers several reactive clients for use with a reactive datasource.
```

quarkus.datasource.reactive.max-size=20
----

===== Reactive connection pool size adjustment

To protect your database from overloading during the transaction load peaks, size the pool adequately to throttle the database load.
The proper size always depends on many factors, such as the number of parallel application users or the nature of the workload.

> Be aware that this pool size adjustment will cause some requests to time out while waiting for a connection.

Reviewer 3: ```suggestion
Be aware that setting the pool size too low may cause some requests to time out while waiting for a connection.
```

quarkus.datasource.reactive.max-size=20
----

===== Reactive connection pool size adjustment

> To protect your database from overloading during the transaction load peaks, size the pool adequately to throttle the database load.

Reviewer 2: ```suggestion
To protect your database from overloading during load peaks, size the pool adequately to throttle the database load.
```

The recommended approach is to use the real database you intend to use in production, especially when xref:databases-dev-services.adoc[Dev Services provide a zero-config database for testing], and running tests against a container is relatively quick and produces expected results on an actual environment.
However, it is also possible to use JVM-powered databases for scenarios when the ability to run simple integration tests is required.

==== Support and limitations

Embedded databases (H2 and Derby) work in JVM mode.
For native mode, the following limitations apply:

> * Derby cannot be embedded into the application in native mode; it only supports a remote connection.
> The Quarkus Derby extension allows native compilation of the Derby JDBC client, supporting only remote connections.

Reviewer 3: This feels like you're saying the same thing twice?

```suggestion
* Derby cannot be embedded into the application in native mode.
However, the Quarkus Derby extension allows native compilation of the Derby JDBC *client*, supporting *remote* connections.
```

Author: Good one! Applying.

> * Derby cannot be embedded into the application in native mode; it only supports a remote connection.
> The Quarkus Derby extension allows native compilation of the Derby JDBC client, supporting only remote connections.

Reviewer 4: ```suggestion
* Derby cannot be embedded into the application in native mode; only a remote connection is supported.
This is because the Quarkus Derby extension only allows native compilation of the Derby JDBC client.
```

Reviewer 4: I would say it like this :-)

The recommended approach is to use the real database you intend to use in production, especially when xref:databases-dev-services.adoc[Dev Services provide a zero-config database for testing], and running tests against a container is relatively quick and produces expected results on an actual environment.
However, it is also possible to use JVM-powered databases for scenarios when the ability to run simple integration tests is required.

==== Support and limitations

Embedded databases (H2 and Derby) work in JVM mode.
For native mode, the following limitations apply:

* Derby cannot be embedded into the application in native mode.
However, the Quarkus Derby extension allows native compilation of the Derby JDBC *client*, supporting *remote* connections.

> * Embedding H2 within your native image is not recommended.

Reviewer: @MichalMaler - Some typos need fixing in the Java source of the generated config prop tables. Not sure if this is in a separate PR:

For example, `a health` not `an health`.
![image](https://github.com/quarkusio/quarkus/assets/92924207/60b9f01a-ecde-4a38-b04d-5741cbeb6980)

Author: I run a massive check across all included javadocs. It is ok now. Thx for spotting! @michelle-purcell
